javagen/src/main/java/com/azure/autorest/template/ServiceClientBuilderTemplate.java

```diff
@@ -340,6 +348,11 @@ protected void addCreateHttpPipelineMethod(JavaSettings settings, JavaClass clas
             function.line("policies.addAll(this.pipelinePolicies);");
             function.line("HttpPolicyProviders.addAfterRetryPolicies(policies);");
 
+            // clientOptions header
+            function.line("HttpHeaders headers = new HttpHeaders();");
+            function.line("clientOptions.getHeaders().forEach(header -> headers.set(header.getName(), header.getValue()));");
+            function.ifBlock("headers.getSize() > 0", block -> block.line("policies.add(new AddHeadersPolicy(headers));"));
```

Comment:
Should we add this before `RetryPolicy`? It seems the headers cannot change per request? Though in current data-plane code it seems always after `RetryPolicy`.

Also it might be need to be before the credential policy.